### PR TITLE
Simplify the public surface design of Response<T> for usability.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1954

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp:46]:
```suggestion
      auto& rawResponse = response.RawResponse;
```
Then update line 49 to `rawResponse.` instead of `rawResponse->`.  I made this change to my latest keyvault PR https://github.com/Azure/azure-sdk-for-cpp/pull/1967/files#diff-bc3dcb1e806ad216a89560228cedd0cb8d51838217308297720d24c90c2d7f94L46

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp:229]: since you are here :)  would you mind?

Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp:229]: What would you like me to do?

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp:46]: Can you please explain why we want to do this?

Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp:46]: I think I get it. You don't want `CheckValidResponse` to extract ownership of the `RawResponse`, because then the caller can't keep using it from the `Response<T>` passed-in.

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp:229]: missing new line

**Review by vhvb1989 — APPROVED** [review]
awesome!

**Review by JeffreyRichter — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp:229]: Ah gotcha. I wasn't seeing it because of differences between PR conversation view (which is what I typically use) and file view on GitHub:
image

Versus:
https://github.com/Azure/azure-sdk-for-cpp/pull/1974/files
image

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp]: > Then update line 49 to `rawResponse.` instead of `rawResponse->`.

This wasn't required.

File: sdk/keyvault/azure-security-keyvault-keys/src/delete_key_operation.cpp
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/src/delete_key_operation.cpp:61]: @vhvb1989, if such a scenario does come up, where you need this check, please let me know and add a test. Looking at existing tests and code coverage, this code-block and exception was never hit.

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]
Please wait for storage team's approval

**Review by Jinming-Hu — APPROVED** [review]
Changes to Storage SDK LGTM

**Comment by ahsonkhan** [comment]
Anyone else have any feedback on this PR before I merge it? @antkmsft  or @RickWinter - does anything significant jump out at you?

File: sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp:290]: @vhvb1989 we may want to rename the `Value` property on `KeyBackup`, since `.Value.Value` reads as quite confusing. It's in `_detail` btw, so not critical:
https://github.com/Azure/azure-sdk-for-cpp/blob/ddd7a093f44cc997d1ffc66fd9a018b43c965e45/sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/details/key_backup.hpp#L20-L23

**Pull request merged**
